### PR TITLE
return correct error on duplicate email while updating user

### PR DESCRIPTION
**Pull request by tinfoil-knight**
Closes: #7249 

Currently, a user updating their email to one that's already in use returns an `500 Internal Server Error`. This PR updates the implementation to ensure that a `400 Bad Request` error with a description is returned instead.

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

- Create a user (using the `POST /users` endpoint)
- Attempt to modify the user (using the `PUT /user/current` endpoint)
- Verify that you're receiving an error with HTTP status code 400 with a message mentioning that the email already exists

## If this is a bugfix, which Traffic Control versions contained the bug?

 
Wasn't able to track down the exact release where this bug originated. This bug is present in `master` branch & https://github.com/apache/trafficcontrol/pull/3996 is the PR where the endpoint was first written for API V3.

## PR submission checklist
- [ ] This PR has tests 
- [ ] This PR has documentation
	PR doesn't need any separate documentation since only the error handling behaviour has been changed. API docs only document the request & response structure.
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Comment by tinfoil-knight** [comment]
Wanted to confirm if a test needs to be added here.

I was looking at the test files & I'm slightly confused as to how/where users are added in the mock database.

The relevant file is here: https://github.com/apache/trafficcontrol/blob/master/traffic_ops/testing/api/v5/user_current_test.go

File: traffic_ops/traffic_ops_golang/user/current.go [hunk 1]
```diff
@@ -479,9 +479,9 @@ func ReplaceCurrent(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if err = updateLegacyUser(&user, tx, changePasswd, changeConfirmPasswd); err != nil {
```
Comment (tinfoil-knight) [inline traffic_ops/traffic_ops_golang/user/current.go:481]: Note for Reviewer: `ReplaceCurrent` method is used for Traffic Ops API V3.

File: traffic_ops/traffic_ops_golang/user/current.go [hunk 2]
```diff
@@ -600,9 +600,9 @@ func ReplaceCurrentV4(w http.ResponseWriter, r *http.Request) {
 	}
```
Comment (tinfoil-knight) [inline traffic_ops/traffic_ops_golang/user/current.go:601]: Note: Note for Reviewer: `ReplaceCurrentV4` method is used for both Traffic Ops API V4 & V5.

**Review by ocket8888 — APPROVED** [review]
Looks good, works, fixes the bug

**Pull request merged**